src/openvpn/buffer.c

```diff
@@ -442,6 +442,7 @@ format_hex_ex (const uint8_t *data, int size, int maxoutput,
 				    ((size * 2) + (size / (space_break_flags & FHE_SPACE_BREAK_MASK)) * (int) strlen (separator) + 2),
 				    gc);
   int i;
+  ASSERT( maxoutput > 0 || separator != NULL );
```

Comment:
This will never happen: If maxoutput < 0, alloc_buf will exit with error, if maxoutput == 0 and separator == NULL the strlen() will segfault. 
If the idea is to avoid possible NULL dereferencing in strlen(), the ASSERT should precede that.